### PR TITLE
[docs] Update Contribution Guidelines

### PR DESCRIPTION
**Pull request by PratikMahajan**
Updated contribtion guidelines to include the /hold workflow

### CODE REVIEW

File: CONTRIBUTION.md [hunk 1]
```diff
@@ -74,11 +74,13 @@ git tools.
   - Ensure each PR commit compiles and all unit and e2e tests pass on your local machine.
     - PRs that have vendor commits are an exception to this rule.
 - Following are the things you should keep in mind once you open a PR:
+  - Add a hold as soon as you open the PR by commenting `\hold`
```
Comment (suhanime) [inline CONTRIBUTION.md]: `/hold`

File: CONTRIBUTION.md [hunk 2]
```diff
@@ -74,11 +74,13 @@ git tools.
...
   - Wait for comments from at least 2 reviewers before pushing changes.
   Open comments in the meanwhile can be worked on locally.
   - PR comments should be addressed in new commits. Before final approval, they have to be squashed.
   - If PR has multiple commits, changes requested should eventually be squashed into the original commit where the
   change was requested before the final merge. Each commit in the final PR before merge should pass the tests and be
   usable. (see below)
+  - Once the PR is approved, remove hold by commenting `\hold cancel` to merge the PR
```
Comment (suhanime) [inline CONTRIBUTION.md]: Do we want to call out that we need to squash before cancelling the hold?
Also, `/hold cancel`

Comment (PratikMahajan) [reply CONTRIBUTION.md]: updated

Comment (PratikMahajan) [reply CONTRIBUTION.md]: updated

File: CONTRIBUTION.md [hunk 3]
```diff
@@ -74,11 +74,13 @@ git tools.
...
 - Following are the things you should keep in mind once you open a PR:
+  - Add a hold as soon as you open the PR by commenting `/hold`
   - Wait for comments from at least 2 reviewers before pushing changes.
   Open comments in the meanwhile can be worked on locally.
   - PR comments should be addressed in new commits. Before final approval, they have to be squashed.
   - If PR has multiple commits, changes requested should eventually be squashed into the original commit where the
-  change was requested before the final merge. Each commit in the final PR before merge should pass the tests and be
+  change was requested before cancelling the hold. Each commit in the final PR before merge should pass the tests and be
   usable. (see below)
+  - Once the PR is approved, remove hold by commenting `/hold cancel` to merge the PR
```
Comment (aravindhp) [inline CONTRIBUTION.md]: `remove the hold by...`

**Comment by mansikulkarni96** [comment]
lgtm

**Pull request merged**